node/impl/full/eth.go

```diff
@@ -474,6 +474,10 @@ func (a *EthModule) EthGetTransactionCount(ctx context.Context, sender ethtypes.
 		return ethtypes.EthUint64(nonce), nil
 	}
 
+	if blkParam.PredefinedBlock != nil && *blkParam.PredefinedBlock == "earliest" {
+		return ethtypes.EthUint64(0), nil
```

Comment:
@virajbhartiya I am not convinced we need this. What happens to the itests if we don't special case the genesis block here ?

Reply:
If we don't add the check for "earliest" specifically it returns an error saying 
```bash
        	Error:      	Received unexpected error:
        	            	failed to process block param: {0x14007a6a750 <nil> <nil> false}; block param "earliest" is not supported: block param "earliest" is not supported
        	Test:       	TestEthGetTransactionCount
```